### PR TITLE
PIM-5642: fix catalog scope filter from the product grid

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

Fix the product grid scope switcher context coming from product edif form

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | NA
| Added Behats                      | Y
| Changelog updated                 | N
| Review and 2 GTM                  | N
| Micro Demo to the PO (Story only) | NA
| Migration script                  | N
| Tech Doc                          | N

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js [hunk 1]
```diff
@@ -24,14 +24,21 @@ define(
...
                 mediator.once('datagrid_filters:rendered', this.moveFilter.bind(this));
+                mediator.once('datagrid_filters:rendered', this.resetValue.bind(this));
 
                 mediator.bind('grid_load:complete', function(collection) {
                     $('#grid-' + collection.inputName).find('div.toolbar').show();
                 });
             },
 
+            /**
+             * Move the filter to it's proper position
```
Comment (grena) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js]: > its

File: features/datagrid/products_backtothegrid.feature [hunk 2]
```diff
@@ -30,6 +30,18 @@ Feature: Products back to the grid
     And I should see product sneakers_1
     And I should not see product boots_1
 
+  Scenario: Successfully restore the scope dropdown
+    Given I filter by "SKU" with value "sneakers_1"
+    And the grid should contain 1 element
+    And I should see the text "E-Commerce"
+    And I should not see the text "Mobile"
```
Comment (pierallard) [inline features/datagrid/products_backtothegrid.feature]: Useful ?

File: features/datagrid/products_backtothegrid.feature [hunk 3]
```diff
@@ -30,6 +30,18 @@ Feature: Products back to the grid
...
+    And I click on the "sneakers_1" row
+    And I switch the scope to "mobile"
+    And I click back to grid
+    Then the grid should contain 1 element
+    And I should see the text "Mobile"
+    And I should not see the text "E-Commerce"
```
Comment (pierallard) [inline features/datagrid/products_backtothegrid.feature]: Useful ?

**Comment by pierallard** [comment]
Just 1 question and GTM.

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js [hunk 4]
```diff
@@ -24,14 +24,21 @@ define(
 
             initialize: function() {
                 SelectFilter.prototype.initialize.apply(this, arguments);
+                this.catalogScope = UserContext.get('catalogScope');
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js]: Please declare it as a property.

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js [hunk 5]
```diff
@@ -44,6 +51,15 @@ define(
             },
 
             /**
+             * Reset the current value with the catalogScope from the usercontext
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/product_scope-filter.js]: Too technical, I still doesn't know what this method does after reading this.

File: features/datagrid/products_backtothegrid.feature (hunk 2 above)
Comment (juliensnz) [inline features/datagrid/products_backtothegrid.feature]: yeah, I think

File: features/datagrid/products_backtothegrid.feature (hunk 3 above)
Comment (juliensnz) [inline features/datagrid/products_backtothegrid.feature]: yeah, also

**Pull request closed without merging**